src/sql/parts/jobManagement/views/jobStepsTree.ts

```diff
@@ -0,0 +1,512 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as DOM from 'vs/base/browser/dom';
+import * as tree from 'vs/base/parts/tree/browser/tree';
+import * as TreeDefaults from 'vs/base/parts/tree/browser/treeDefaults';
+import { Promise, TPromise } from 'vs/base/common/winjs.base';
+import { IMouseEvent } from 'vs/base/browser/mouseEvent';
+import { generateUuid } from 'vs/base/common/uuid';
+import { JobManagementUtilities } from 'sql/parts/jobManagement/common/jobManagementUtilities';
+import * as Model from 'vs/base/parts/tree/browser/treeModel';
+import { Color } from 'vs/base/common/color';
+import { INavigator, MappedNavigator } from 'vs/base/common/iterator';
+import { Event, Emitter, Relay } from 'vs/base/common/event';
+import { TreeContext } from 'vs/base/parts/tree/browser/treeImpl';
+import { mixin } from 'vs/base/common/objects';
+import { JobStepsTreeView } from 'sql/parts/jobManagement/views/jobStepsTreeView';
+
+const defaultStyles: tree.ITreeStyles = {
+	listFocusBackground: Color.fromHex('#073655'),
+	listActiveSelectionBackground: Color.fromHex('#0E639C'),
+	listActiveSelectionForeground: Color.fromHex('#FFFFFF'),
+	listFocusAndSelectionBackground: Color.fromHex('#094771'),
+	listFocusAndSelectionForeground: Color.fromHex('#FFFFFF'),
+	listInactiveSelectionBackground: Color.fromHex('#3F3F46'),
+	listHoverBackground: Color.fromHex('#2A2D2E'),
+	listDropBackground: Color.fromHex('#383B3D')
+};
+
+export class JobStepsTree implements tree.ITree {
```

Comment:
is there a reason you can't derive from `Tree` instead of reimplementing?  It seems like we are copy\paste duplicating a lot of code?

Reply:
I can't extend the tree because Typescript doesn't allow overriding private methods and the scrolling is all done by the view and scrollable elements, which are private to the tree.